### PR TITLE
pr split nine - add 10 files

### PR DESCRIPTION
**Pull request by michaelkernaghan**
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: integration-tests/contract-with-bad-code.spec.ts [hunk 1]
```diff
@@ -3,13 +3,13 @@ import { badCode } from "./data/badCode";
...
   const Tezos = lib;
-  describe(`Test contract failing with bad code using: ${rpc}`, () => {
+  describe(`Test contract origination fail with bad code through contract api using: ${rpc}`, () => {
 
     beforeEach(async (done) => {
       await setup()
       done()
     })
-    it('fails with a 400 RPC response', async (done) => {
+    it('Verify that contract.originate for a contract with 400 in code will return 400 RPC in response', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/contract-with-bad-code.spec.ts:12]: We could use the same definition as the one in the corresponding wallet test:
`Verify that contract.originate for a contract with a missing instruction (FAILWITH_TYPO) in code will return 400 RPC in response`

File: integration-tests/ledger-signer-failing-tests.spec.ts [hunk 2]
```diff
@@ -20,8 +20,8 @@ CONFIGS().forEach(({ lib, setup }) => {
             done();
         });
 
-        describe('Tests where user declines all prompts on the ledger device', () => {
-            it('Should throw error when user declines to provide public key', async (done) => {
+        describe('Test to verify that user can decline all prompts on the ledger device', () => {
+            it('As a User I want to verify that Ledger will throw an error when I declined to provide public key', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer-failing-tests.spec.ts:24]: Should we use the same format as the other files "Verify that..." without "As a User I want to"?

File: integration-tests/ledger-signer-failing-tests.spec.ts [hunk 3]
```diff
@@ -37,7 +37,7 @@ CONFIGS().forEach(({ lib, setup }) => {
                 done();
             });
 
-            it('Should throw error when user declines to sign', async (done) => {
+            it('As a User I want to verify that Ledger will throw an error when I declined to sign', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer-failing-tests.spec.ts:40]: Should we use the same format as the other files "Verify that..." without "As a User I want to"

File: integration-tests/ledger-signer-failing-tests.spec.ts [hunk 4]
```diff
@@ -55,9 +55,9 @@ CONFIGS().forEach(({ lib, setup }) => {
                 done();
             });
 
-            describe('Should be abble to used Ledger with wallet API', () => {
+            describe('Test that Ledger can be used with wallet API', () => {
                 jest.setTimeout(60000)
-                it('Should throw error when user declines transaction with Ledger', async (done) => {
+                it('SAs a User I want to verify that Ledger will throw an error when I declined to originate contract with Ledger\n', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer-failing-tests.spec.ts:60]: Should we use the same format as the other files "Verify that..." without "As a User I want to"

File: integration-tests/ledger-signer.spec.ts [hunk 5]
```diff
@@ -36,15 +36,15 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
       done();
     });
 
-    it('LedgerSigner is instantiable with default parameters', () => {
+    it('As a User I want to verify that LedgerSigner is instantiable with default parameters', () => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:39]: Should we use the same format as the other files "Verify that..." without "As a User I want to"

File: integration-tests/ledger-signer.spec.ts [hunk 6]
```diff
@@ -36,15 +36,15 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
...
       expect(
         new LedgerSigner(
           transport
         )
       ).toBeInstanceOf(LedgerSigner);
     });
 
-    it('LedgerSigner is instantiable with parameters', () => {
+    it('As a User I want to verify that LedgerSigner is instantiable with parameters', () => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:47]: Should we use the same format as the other files "Verify that..." without "As a User I want to"

File: integration-tests/ledger-signer.spec.ts [hunk 7]
```diff
@@ -55,8 +55,8 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
       ).toBeInstanceOf(LedgerSigner);
     });
 
-    describe('Get the public key', () => {
-      it('Should get the correct public key and public key hash of the Ledger for tz1 curve and default path', async (done) => {
+    describe('Test retrieving the public key from the Ledger', () => {
+      it('As a User I want to verify that Ledger will provide correct public key and public key hash for tz1 curve and default path', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:59]: Should we use the same format as the other files "Verify that..." without "As a User I want to"

File: integration-tests/ledger-signer.spec.ts [hunk 8]
```diff
@@ -74,7 +74,7 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
         done();
       });
 
-      it('Should get the correct public key and public key hash of the Ledger for tz2 curve and default path', async (done) => {
+      it('As a User I want to verify that Ledger will provide correct public key and public key hash for tz2 curve and default path', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:77]: Should we use the same format as the other files "Verify that..." without "As a User I want to"

File: integration-tests/ledger-signer.spec.ts [hunk 9]
```diff
@@ -92,7 +92,7 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
         done();
       });
 
-      it('Should get the correct public key and public key hash of the Ledger for tz3 curve and path having 1 as account value', async (done) => {
+      it('As I User I want to verify that that Ledger will provide correct public key and public key hash for tz3 curve and path having 1 as account value', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:95]: Should we use the same format as the other files "Verify that..." without "As a User I want to"

File: integration-tests/ledger-signer.spec.ts [hunk 10]
```diff
@@ -111,9 +111,9 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
       });
     });
 
-    describe('Should sign operation with Ledger', () => {
+    describe('Test sign operation with Ledger', () => {
       jest.setTimeout(30000);
-      it('Should return the correct signature with the Ledger', async (done) => {
+      it('As a User I want to verify that Ledger returns the correct signature with the Ledger', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:116]: Should we use the same format as the other files "Verify that..." without "As a User I want to"

File: integration-tests/ledger-signer.spec.ts [hunk 11]
```diff
@@ -137,9 +137,9 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
       });
     })
 
-    describe('Should be able to use Ledger with contract API', () => {
+    describe('Test an ability to use Ledger with contract api', () => {
       jest.setTimeout(240000)
-      it('Should originate contract with Ledger', async (done) => {
+      it('As a User I want to verify that I can originate contract with Ledger', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:142]: Should we use the same format as the other files "Verify that..." without "As a User I want to"

File: integration-tests/ledger-signer.spec.ts [hunk 12]
```diff
@@ -164,10 +164,10 @@ CONFIGS().forEach(({ lib, setup, rpc }) => {
       });
     })
 
-    describe('Should be able to used Ledger with wallet API', () => {
+    describe('Test an ability to use Ledger with wallet api', () => {
       jest.setTimeout(120000)
 
-      it('Should sign and inject transaction with Ledger', async (done) => {
+      it('As a User I should be able to sign and inject transaction with Ledger', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer.spec.ts:170]: Should we use the same format as the other files "Verify that..." without "As a User I want to"

**Review by roxaneletourneau — CHANGES_REQUESTED** [review]

File: integration-tests/contract-tx-wait-2-confirmations.spec.ts [hunk 13]
```diff
@@ -2,14 +2,15 @@ import { CONFIGS } from "./config";
 
 CONFIGS().forEach(({ lib, rpc, setup }) => {
   const Tezos = lib;
-  describe(`Test tx and waiting for 2 confirmations using: ${rpc}`, () => {
+  const test = require('jest-retries');
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply integration-tests/contract-tx-wait-2-confirmations.spec.ts:5]: removed

File: integration-tests/contract-unit-as-param.spec.ts [hunk 14]
```diff
@@ -3,15 +3,15 @@ import { depositContractCode, depositContractStorage } from "./data/deposit_cont
 
 CONFIGS().forEach(({ lib, rpc, setup }) => {
   const Tezos = lib;
+  const test = require('jest-retries');
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply integration-tests/contract-unit-as-param.spec.ts:6]: removed

File: integration-tests/contract-with-bad-code.spec.ts (hunk 1 above)
Comment (michaelkernaghan) [reply integration-tests/contract-with-bad-code.spec.ts:12]: changed to 'Verify that contract.originate for a contract with a missing instruction (FAILWITH_TYPO) in code will return 400 RPC in response'

File: integration-tests/ledger-signer-failing-tests.spec.ts (hunk 2 above)
Comment (michaelkernaghan) [reply integration-tests/ledger-signer-failing-tests.spec.ts:24]: Changed to:
'Verify that Ledger will throw an error when no public key is provided'

File: integration-tests/ledger-signer-failing-tests.spec.ts (hunk 3 above)
Comment (michaelkernaghan) [reply integration-tests/ledger-signer-failing-tests.spec.ts:40]: changed to
'Verify that Ledger will throw an error when when signing is declined'

File: integration-tests/ledger-signer-failing-tests.spec.ts (hunk 4 above)
Comment (michaelkernaghan) [reply integration-tests/ledger-signer-failing-tests.spec.ts:60]: changed to
'Verify that Ledger will throw an error when contract origination is declined'

File: integration-tests/ledger-signer.spec.ts (hunk 6 above)
Comment (michaelkernaghan) [reply integration-tests/ledger-signer.spec.ts:47]: Changed to
'Verify that LedgerSigner is instantiable with parameters'

File: integration-tests/ledger-signer.spec.ts (hunk 8 above)
Comment (michaelkernaghan) [reply integration-tests/ledger-signer.spec.ts:77]: Changed to
'Verify that Ledger will provide correct public key and public key hash for tz1 curve and default path'

File: integration-tests/ledger-signer.spec.ts (hunk 10 above)
Comment (michaelkernaghan) [reply integration-tests/ledger-signer.spec.ts:116]: Changed to
'Verify that that Ledger will provide correct public key and public key hash for tz3 curve and path having 1 as account value'

File: integration-tests/ledger-signer.spec.ts (hunk 11 above)
Comment (michaelkernaghan) [reply integration-tests/ledger-signer.spec.ts:142]: changed to
'Verify that Ledger returns the correct signature'

File: integration-tests/ledger-signer-failing-tests.spec.ts [hunk 15]
```diff
@@ -55,9 +56,11 @@ CONFIGS().forEach(({ lib, setup }) => {
                 done();
             });
 
-            describe('Should be abble to used Ledger with wallet API', () => {
+            describe('Test that Ledger can be used with wallet API', () => {
+                
                 jest.setTimeout(60000)
-                it('Should throw error when user declines transaction with Ledger', async (done) => {
+
+                it('Verify that Ledger will throw an error when contract origination is declined', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer-failing-tests.spec.ts:63]: The operation is a transaction, not orgination

Comment (michaelkernaghan) [reply integration-tests/ledger-signer-failing-tests.spec.ts:63]: changed to
Verify that Ledger will throw an error when a transaction is declined

File: integration-tests/ledger-signer.spec.ts (hunk 5 above)
Comment (roxaneletourneau) [reply integration-tests/ledger-signer.spec.ts:39]: @michaelkernaghan

File: integration-tests/ledger-signer.spec.ts (hunk 12 above)
Comment (roxaneletourneau) [reply integration-tests/ledger-signer.spec.ts:173]: @michaelkernaghan

File: integration-tests/ledger-signer-failing-tests.spec.ts [hunk 16]
```diff
@@ -37,7 +38,7 @@ CONFIGS().forEach(({ lib, setup }) => {
                 done();
             });
 
-            it('Should throw error when user declines to sign', async (done) => {
+            it('Verify that Ledger will throw an error when when signing is declined', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/ledger-signer-failing-tests.spec.ts:41]: ~when~ when

File: integration-tests/ledger-signer.spec.ts (hunk 5 above)
Comment (michaelkernaghan) [reply integration-tests/ledger-signer.spec.ts:39]: changed to : Verify signing and injecting a transaction with Ledger

File: integration-tests/ledger-signer.spec.ts (hunk 7 above)
Comment (michaelkernaghan) [reply integration-tests/ledger-signer.spec.ts]: Updated the examples in this test set that were not "verify .." format

File: integration-tests/ledger-signer-failing-tests.spec.ts (hunk 16 above)
Comment (michaelkernaghan) [reply integration-tests/ledger-signer-failing-tests.spec.ts:41]: fixed

**Review by roxaneletourneau — APPROVED** [review]

**Pull request merged**